### PR TITLE
Release dataproc 0.4.0

### PR DESCRIPTION
**Pull request by sduskis**
This pull request was generated using releasetool.

### CODE REVIEW

File: dataproc/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,29 @@
...
+05-30-2019 05:52 PDT
+
+### Implementation Changes
+- Update docs/conf.py, add routing header to method metadata, fix docstrings (via synth). ([#7924](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/7924))
+
+### New Features
+- Add new service features for v1, including autoscaling (via synth). ([#8152](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/8152))
+- Add new service features for v1beta2, including autoscaling (via synth). ([#8119](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/8119))
+
+### Dependencies
```
Comment (busunkim96) [inline dataproc/CHANGELOG.md]: Would you mind removing this section since it's empty?

**Comment by sduskis** [comment]
@busunkim96, what do we need to do to move the dataproc release forward?

**Comment by busunkim96** [comment]
It doesn't look like the key changes will land today. I'll take over and complete it manually.

**Comment by busunkim96** [comment]
Tagged: https://github.com/googleapis/google-cloud-python/releases/tag/dataproc-0.4.0

**Comment by busunkim96** [comment]
@sduskis Could I get your review on #8204?

**Comment by busunkim96** [comment]
Released: https://pypi.org/project/google-cloud-dataproc/0.4.0

**Pull request merged**